docs/database/main.md

```diff
@@ -0,0 +1,105 @@
+# Database Specifications
+
+This documented describes the type of data that is stored in our MongoDB
+cluster.
+
+## Account
+
+```json
+{
+  "id": "String",
+  "rscid": "String",
+  "created": "Date",
+  "username": "String",
+  "password": "encrypt<String>"
+}
+```
+
+## Semester
+
+```json
+{
+  "id": "String",
+  "year": "Number",
+  "type": "'fall' | 'spring' | 'summer'"
+}
+```
+
+## Major
+
+```json
+{
+  "id": "String",
+  "department": "String",
+  "code": "String",
+  "name": "String",
+}
+```
+
+## Class
+
+```json
+{
+  "id": "String",
+  "major": "ref<Major>",
+  "semester": "ref<Semester>",
+  "code": "String",
+}
+```
+
+## Section
+
+```json
+{
+  "id": "String",
+  "class": "ref<Class>",
+  "professors": "List<String>",
+  "section": "Number",
+}
+```
+
+## Note
+
+```json
+{
+  "id": "String",
+  "name": "String",
+  "description?": "String",
+  "created": "Date",
+  "type": "'link' | 'crib-sheet' | 'notes' | ''",
+  "file": {
+    "type": "'pdf' | 'png' | 'webp' | 'docx' | 'mp3'",
+    "path": "String",
+    "internal": "Boolean"
+  },
+  "section": "ref<Section>",
+  "likes": "set<ref<Account>>",
+  "dislikes": "set<ref<Account>>",
+}
+```
+
+## Report
+
+```json
+{
+  "id": "String",
+  "type": "'note' | 'comment'",
+  "note": "ref<Note>",
+  "comment": "ref<Comment>"
+}
+```
+
+## Comment
+
+```json
+{
+  "id": "String",
+  "comment": "String",
+  "created": "Date",
+  "account": "ref<Account>",
+  "note": "ref<Note>",
+  "likes": "set<ref<Account>>",
+  "dislikes": "set<ref<Account>>",
```

Comment:
If you mean why a set and not a list, because all of the items must be unique. But in MongoDB, it would be represented all the same internally: a list.

If you mean why a set or list at all, it is because in MongoDB it is harder to do `JOIN` commands. So, rather an having a table for likes and dislikes, just having them on each document and account is faster.